File: docs/guides/csharp/src/setHeaderUserIDThenSaveObjects.cs
namespace Algolia;

using System;
using System.Collections.Generic;
using System.Net.Http;
using System.Text.Json;
using Algolia.Search.Clients;
using Algolia.Search.Http;
using Algolia.Search.Models.Search;

class SetHeaderUserIDThenSaveObjects
{
  private static readonly List<Dictionary<string, object>> playlists = []; // Your records

  async Task Main(string[] args)
  {
    try
    {
      var client = new SearchClient(new SearchConfig("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY"));
      foreach (var playlist in playlists)
      {
        var playlistUserID = playlist.GetValueOrDefault("userID", "") as string;
        await client.SaveObjectsAsync(
          "<YOUR_INDEX_NAME>",
          playlists,
          false,
          1000,
          new RequestOptionBuilder().AddExtraHeader("X-Algolia-User-ID", playlistUserID).Build()
        );
      }
    }
    catch (Exception e)
    {
      Console.WriteLine(e.Message);
    }
  }
}

File: docs/guides/swift/Sources/setHeaderUserIDThenSaveObjects.swift
import Foundation
#if os(Linux) // For linux interop
    import FoundationNetworking
#endif

import Core
import Search

func setHeaderUserIDThenSaveObjects() async throws {
    let playlists: [[String: AnyCodable]] = [ /* Your records */ ]

    let client = try SearchClient(appID: "ALGOLIA_APPLICATION_ID", apiKey: "ALGOLIA_API_KEY")

    for playlist in playlists {
        do {
            let playlistUserID = playlist["userID"]?.value as! String
            try await client.saveObjects(
                indexName: "<YOUR_INDEX_NAME>",
                objects: playlists,
                waitForTasks: false,
                batchSize: 1000,
                requestOptions: RequestOptions(
                    headers: ["X-Algolia-User-ID": playlistUserID]
                )
            )
        } catch {
            print(error)
        }
    }
}
